docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,104 @@
+# Architecture of Crates.io
+
+This document is an intro to the codebase in this repo. If you want to work on a bug or a feature,
+hopefully after reading this doc, you'll have a good idea of where to start looking for the code
+you want to change.
+
+This is a work in progress. Pull requests and issues to improve this document are very welcome!
+
+## Documentation
+
+Documentation about the codebase appears in these locations:
+
+* `LICENSE-APACHE` and `LICENSE-MIT` - the terms under which this codebase is licensed.
```

Comment:
It looks like `package.json` only lists MIT.  I'm not sure if that is an oversight or if we need to word this more carefully.  We should probably explicitly call out dual licensing in `README.md` 

Also see: https://docs.npmjs.com/files/package.json#license